docs/typescript.md

```diff
@@ -2,89 +2,70 @@
+Because Gasket is a plugin framework, plugins themselves are responsible for augmenting the core Gasket interfaces. This means that the type declarations for the plugins must be "discovered" in your TypeScript codebase. You can ensure this by referencing or importing your presets or plugins. The sample code snippets below show these imports, but if you can centralize the imports in one area of your codebase, you can avoid duplication and ensure that TypeScript is aware of all the plugins.
 
-## Validating gasket.config.js
+## Writing and Validating Gasket Configuration
 
-The `@gasket/engine` package supplies a `GasketConfigFile` type that, once you've also imported all your plugins, validates the contents of your gasket config file.
+In Gasket, you can now write your `gasket.js` configuration file as `gasket.ts` using TypeScript. This allows for full type-checking and validation directly within your TypeScript environment.
 
-The `gasket.config.js` file cannot be written in TypeScript, but you can configure TypeScript to check JS files or use a `@ts-check` comment along with JSDoc type annotations. Due to JSDoc type checking limitations, you may have to separate the config declaration and your export:
+The `@gasket/core` package supplies a `GasketConfigFile` type that validates the contents of your Gasket config file. When you import all your plugins into `gasket.ts`, TypeScript can fully validate the configuration.
 
-```javascript
-//@ts-check
-///<reference types="@gasket/preset-nextjs"/>
-
-/** @type {import('@gasket/engine').GasketConfigFile} */
-const config = {
-  plugins: {
-    presets: ['@gasket/nextjs']
-  },
-  compression: true,
-  http: 8080,
-  intl: {
-    defaultLocale: 'en-GB'
-  }
-};
-
-module.exports = config;
-```
-
-## Validating lifecycle scripts
-
-The `@gasket/engine` package supplies a `Hook` type that can be used to validate lifecycle scripts and plugin hooks. It takes a string type parameter for the name of the lifecycle.
+Here's an example:
 
 ```typescript
-import { Hook } from '@gasket/engine';
-import '@gasket/plugin-intl';
-
-const intlLocaleHandler: Hook<'intlLocale'> = (gasket, locale, { req, res }) => {
-  // return 3; - does not pass validation
-  return getLocaleFromHost(req.headers.host);
-}
+// gasket.ts
+import type { GasketConfigDefinition } from '@gasket/core';
+import { makeGasket } from '@gasket/core';
+import pluginA from '@gasket/plugin-a';
+import pluginB from '@gasket/plugin-b';
+
+export default makeGasket({
+  plugins: [
+    pluginA,
+    pluginB
+  ]
+  // additional config
+} as GasketConfigDefinition);
```

Comment:
You shouldn't have to explicitly cast the config when configuring in the `makeGasket` argument.

Reply:
Removed & removed from the `write-gasket-config`